### ISSUE TITLE
Backends flicker on setting same value

### ISSUE THREAD

**Issue by Therzok** [mono/xwt#672]
Take a Label for example. If the text is `A` and we do `label.Text = "A"`, the text will flicker. This is caused by propagating the update throughout the whole stack.

The question is: Whose responsibility is it to guard against this? Should this be done at Xwt namespace level or by the user? I think it's safe to be done at Xwt level, as setting the same value should have no side-effects.

**Comment by sevoku** [comment]
I think it depends on the property. `Label.Text` is simple and it could be checked this way, but we should make sure that we don't invoke into native to do this (the label text should be cached in this case, like its done for `Label.Markup`). But there are some other properties which can change from the backend side (like sizes for example).

It would be also great if Widgets would implement `INotifyPropertyChanged`...

**Comment by Therzok** [comment]
Yeah, that's why I put the issue up for debate, as a lot of places have different behaviours.
